lib/bridge/IrcHandler.js

```diff
@@ -538,15 +538,82 @@ IrcHandler.prototype.onMode = Promise.coroutine(function*(req, server, channel,
                                                 mode, enabled, arg) {
     let privateModes = ["k", "i", "s"];
 
-    if (privateModes.indexOf(mode) === -1) {
-        return; // ignore everything but privateModes
-    }
     req.log.info(
         "onMode(%s) in %s by %s (arg=%s)",
         (enabled ? ("+" + mode) : ("-" + mode)),
         channel, by, arg
     );
 
+    if (privateModes.indexOf(mode) !== -1) {
+        req.log.info("Private mode");
+        yield this._onPrivateMode(req, server, channel, by, mode, enabled, arg);
+        return;
+    }
+
+    const matrixRooms = yield this.ircBridge.getStore().getMatrixRoomsForChannel(server, channel);
+
+    if (matrixRooms.length === 0) {
+        req.log.info("No mapped matrix rooms for IRC channel %s", channel);
+        return;
+    }
+
+    const bridgedClient = this.ircBridge.getClientPool().getBridgedClientByNick(
+        server, arg
+    );
+
+    if (!bridgedClient) {
+        req.log.info("No bridged client for %s", arg);
+        return;
+    }
+
+    if (!bridgedClient.unsafeClient) {
+        req.log.info(`Bridged client for ${arg} has no IRC client.`);
+        return;
+    }
+
+    // Bridge operator modes to power levels
+    let modeToPower = server.getModePowerMap();
+
+    if (Object.keys(modeToPower).indexOf(mode) === -1) {
+        // Not an operator power mode
+        return;
+    }
+
+    const userPrefixes = bridgedClient.unsafeClient.chanData(channel).users[arg];
```

Comment:
Dangerous. Don't assume there will always be chanData for the given `channel`. Remember that some other random client may have invoked this `onMode`.